src/schema/workflow_execution_activity.yaml

```diff
@@ -485,19 +516,57 @@ slots:
       TODO      
 
   has_calibration:
-    # domain: ChromatographicSeparationProcess
-    # range: DataObject
-    # description: An instrument data that is used to calibrate retention time or to calculate rentention index
-    range: string
-    description: >-
-      A reference to a file that holds calibration information.
+    any_of:
+      - range: Calibration
+      - range: string
```

Comment:
Why is the range both Calibration and String? having the value of has_calibration in the current schema has been problematic as it is currently populated with both boolean values (false, False) and data object identifiers.

Reply:
I see the notes from below, I recommend writing the slot how it should be and Brynn has has been handling the migrations. So please update this to just Calibration and note in the PR that a migration is needed.  cc @brynnz22

Reply:
Thanks @aclum for the review.  I think we have to wait to do a migration until there is an ingestion, but I may be misunderstanding.  @brynnz22 you've explained this challenge to me before, but I _still_ struggle with the dependencies - can you explain again here?

Reply:
Yes, the range needs to have both `Calibration` and `string` because it is currently `string` for the nmdc production schema. But as discussed in one of the metadata meetings (a few weeks ago), we decided to keep the `has_calibration` slot that was created in Berkeley both on the new classes and the old class of `DataObject` (in the nmdc production schema), because the migration would require that data actually exists on the new classes (e.g. `ChromatographicSeparationProcess`). Its a chicken and egg situation. So we are keeping the old modeling and adding the new modeling to work around this issue. After an ingest of data into the new class (which is also retrospective data in a way), we will then remove the old modeling. In this case, remove the range of `string` and do the migration. Sorry that's confusing to write, happy to talk about it on a call.

Reply:
This issue was created to add the `has_calibration` back into the Berkeley schema after this discussion: https://github.com/microbiomedata/berkeley-schema-fy24/pull/107, so I added the old modeling back in the Berekeley schema.